cmd/strelaysrv/listener.go

```diff
@@ -119,6 +119,15 @@ func protocolConnectionHandler(tcpConn net.Conn, config *tls.Config) {
 
 			switch msg := message.(type) {
 			case protocol.JoinRelayRequest:
+				if msg.Token != token {
```

Comment:
Perhaps it shouldn't be checked if its empty?

Reply:
do you mean in place by `if len(token) > 0 && msg.Token != token` or by restructuring the code in a way that would never need to check it, e.g by introducing a new message-type `JoinRelayRequestWithToken` or something like that?

Reply:
As in, now if you don't set a token, but someone provides one, it will fail.

I don't think providing auth material when none is required should result in a failure?